### ISSUE TITLE
Validation logging with multiple datasets and `add_dataloader_idx=False`

### ISSUE THREAD

**Issue by toshiks** [Lightning-AI/lightning#11680]
## 🐛 Bug

In my case need get value of loss in logs across multiply datasets for drawing on one plot. To achieve that I'm using `add_dataloader_idx=False` when log loss. 

```python
    def validation_step(self, batch, batch_idx, dataloader_idx=0):
        """Validation loop for one batch."""
        predicted = self.forward(batch)
        loss = self._calc_loss_on_predicts(batch, predicted)

        self.log("val/loss", loss, add_dataloader_idx=False)
```

It works well in `pytorch-lightning=1.3.2`. But after upgrading to `1.5.9` it was crashed with error: 

```
File ".../python3.8/site-packages/pytorch_lightning/trainer/connectors/logger_connector/result.py", line 489, 
in log  raise MisconfigurationException(
pytorch_lightning.utilities.exceptions.MisconfigurationException: You called 
`self.log(val/loss, ...)` twice in `validation_step` with different arguments. This is not allowed
```

### Expected behavior

Framework should log value accross multiply datasets without error.

### Environment

* CUDA:
        - GPU:
                - GeForce RTX 2070
                - GeForce RTX 2070
        - available:         True
        - version:           10.2
* Packages:
        - numpy:             1.19.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.2
        - pytorch-lightning: 1.5.9
        - tqdm:              4.50.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.5
        - version:           106-Ubuntu SMP Thu Jan 6 23:58:14 UTC 2022

### Additional context

In my opinion problem in comparison in 488 line here: 

https://github.com/PyTorchLightning/pytorch-lightning/blob/86b177ebe5427725b35fde1a8808a7b59b8a277a/pytorch_lightning/trainer/connectors/logger_connector/result.py#L468-L489

Because changed `dataloader_idx`, `meta` also changed. But in my case it's wrong, isn't it?

cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

**Comment by carmocca** [comment]
Can you try installing master or nightly? We've made several changes to this logic in particular.

**Comment by rohitgr7** [comment]
> self.log("val/loss", loss, add_dataloader_idx=False)

isn't the error correct?
since it tracks the log internally based on the key, and if you set `add_dataloader_idx=False`, for multiple dataloaders, it won't be able to distinguish for which dataloader_idx, key belongs to.

**Comment by carmocca** [comment]
I guess this is asking for https://github.com/PyTorchLightning/pytorch-lightning/issues/11126 to be implemented

**Comment by toshiks** [comment]
> Can you try installing master or nightly? We've made several changes to this logic in particular.

Yep! I'll try it today later.

**Comment by wolframalexa** [comment]
I am also having this issue, but can't find how to install the nightly build. I'm currently using `pip install pytorch-lightning` to install and using pytorch-lightning 1.5.9 on Google Colab. Have you been able to resolve this issue?

**Comment by toshiks** [comment]
So, I intall nightly version by following command: 
`pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/master.zip`

But it wasn't fix situation. I got same `MisconfigurationException`.

Environment

* CUDA:
	- GPU:
		- GeForce RTX 2070
		- GeForce RTX 2070
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.21.2
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.2
	- pytorch-lightning: 1.6.0dev
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.5
	- version:           106-Ubuntu SMP Thu Jan 6 23:58:14 UTC 2022

**Comment by rohitgr7** [comment]
sorry, I am a little confused here. How are we planning to handle cases where these logged values are used. For eg with ModelCheckpoint, EarlyStopping, ReducelLRonPlateau, if both the dataloaders have the same log keys?

**Comment by carmocca** [comment]
The callbacks will not support it, as they rely on the `*_metrics` dictionaries that don't support collisions.

Only one value would be available. The concern is about how it's computed when multiple dataloaders log it.

**Comment by rohitgr7** [comment]
> The concern is about how it's computed when multiple dataloaders log it.

yeah.. this was my question exactly.. haha :)

**Comment by carmocca** [comment]
Closing in favor of https://github.com/Lightning-AI/lightning/issues/11126

**Comment by yanzhiHu996** [comment]
> ## 🐛 Bug
> In my case need get value of loss in logs across multiply datasets for drawing on one plot. To achieve that I'm using `add_dataloader_idx=False` when log loss.
> 
> ```python
>     def validation_step(self, batch, batch_idx, dataloader_idx=0):
>         """Validation loop for one batch."""
>         predicted = self.forward(batch)
>         loss = self._calc_loss_on_predicts(batch, predicted)
> 
>         self.log("val/loss", loss, add_dataloader_idx=False)
> ```
> 
> It works well in `pytorch-lightning=1.3.2`. But after upgrading to `1.5.9` it was crashed with error:
> 
> ```
> File ".../python3.8/site-packages/pytorch_lightning/trainer/connectors/logger_connector/result.py", line 489, 
> in log  raise MisconfigurationException(
> pytorch_lightning.utilities.exceptions.MisconfigurationException: You called 
> `self.log(val/loss, ...)` twice in `validation_step` with different arguments. This is not allowed
> ```
> 
> ### Expected behavior
> Framework should log value accross multiply datasets without error.
> 
> ### Environment
> * CUDA:
>   - GPU:
>   - GeForce RTX 2070
>   - GeForce RTX 2070
>   - available:         True
>   - version:           10.2
> * Packages:
>   - numpy:             1.19.1
>   - pyTorch_debug:     False
>   - pyTorch_version:   1.10.2
>   - pytorch-lightning: 1.5.9
>   - tqdm:              4.50.2
> * System:
>   - OS:                Linux
>   - architecture:
>   - 64bit
>   - ELF
>   - processor:         x86_64
>   - python:            3.8.5
>   - version:           106-Ubuntu SMP Thu Jan 6 23:58:14 UTC 2022
> 
> ### Additional context
> In my opinion problem in comparison in 488 line here:
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/86b177ebe5427725b35fde1a8808a7b59b8a277a/pytorch_lightning/trainer/connectors/logger_connector/result.py#L468-L489
> 
> Because changed `dataloader_idx`, `meta` also changed. But in my case it's wrong, isn't it?
> 
> cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

have you fix this bug?

**Comment by nicolasfischoeder** [comment]
*BUMP*@carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

Can somebody offer a solution please? I ran into the same issue.

**Comment by thangld201** [comment]
@williamFalcon Hi do you have any updates or quick fix suggestions on this ? I'm also facing the same issue. I have to aggregate the same metric value across dataloaders so this is necessary. 
I tried setting 
```
self.log("val/loss",loss,on_step=False,on_epoch=True,sync_dist=self.cfg.get('strategy')!=None,add_dataloader_idx=False)
```
But I still get
```
MisconfigurationException: You called `self.log(val/loss, ...)` twice in `validation_step` with different arguments. This is not
allowed
```

**Comment by drusmanbashir** [comment]
Any luck here? I have the same error, but only when the model is compiled. It logs fine if i dont compile the model!

**Comment by guarin** [comment]
> Any luck here? I have the same error, but only when the model is compiled. It logs fine if i dont compile the model!

Same issue for me, only happens with compiled model.
